internal/tools/aws/database_multitenant.go

```diff
@@ -0,0 +1,730 @@
+	installationID string
+}
+
+// NewRDSMultitenantDatabase returns a new instance of RDSMultitenantDatabase that implements database interface.
+func NewRDSMultitenantDatabase(installationID string, client *Client) *RDSMultitenantDatabase {
+	return &RDSMultitenantDatabase{
+		client:         client,
+		installationID: installationID,
+	}
+}
+
+// Teardown removes all AWS resources related to a RDS multitenant database.
+func (d *RDSMultitenantDatabase) Teardown(store model.InstallationDatabaseStoreInterface, keepData bool, logger log.FieldLogger) error {
+	databaseName := MattermostRDSDatabaseName(d.installationID)
+
+	logger = logger.WithField("rds-multitenant-database", databaseName)
+	logger.Info("Tearing down RDS database and database secret")
+
+	multitenantDatabases, err := store.GetMultitenantDatabases(&model.MultitenantDatabaseFilter{
+		InstallationID: d.installationID,
+	})
+	if err != nil {
+		return errors.Wrapf(err, "cannot teardown multitenant database for installation ID %s", d.installationID)
+	}
+	if len(multitenantDatabases) > 1 {
+		return errors.Errorf("more than one multitenant database per installation is not supported by this database (found %d)", len(multitenantDatabases))
+	}
+	if len(multitenantDatabases) == 0 {
+		logger.Warnf("Installation ID %s does not belong to any of the multitenant databases. Nothing to be torn down.", d.installationID)
+
+		return nil
+	}
+
+	err = d.removeRDSDatabase(multitenantDatabases[0], databaseName, store, logger)
+	if err != nil {
+		return errors.Wrapf(err, "unable to finish teardown of multitenant database name %s", databaseName)
+	}
+
+	multitenantDatabaseSecretName := RDSMultitenantSecretName(d.installationID)
+
+	_, err = d.client.Service().secretsManager.DeleteSecret(&secretsmanager.DeleteSecretInput{
+		SecretId: aws.String(multitenantDatabaseSecretName),
+	})
+	if err != nil && !IsErrorCode(err, secretsmanager.ErrCodeResourceNotFoundException) {
+		return errors.Wrapf(err, "failed to delete multitenant database secret name %s", multitenantDatabaseSecretName)
+	}
+
+	logger.Infof("Multitenant RDS database teardown complete")
+
+	return nil
+}
+
+// Snapshot creates a snapshot of single RDS multitenant database.
+func (d *RDSMultitenantDatabase) Snapshot(store model.InstallationDatabaseStoreInterface, logger log.FieldLogger) error {
+	return errors.New("not implemented")
+}
+
+// GenerateDatabaseSpecAndSecret creates the k8s database spec and secret for accessing a single database inside
+// a RDS multitenant cluster.
+func (d *RDSMultitenantDatabase) GenerateDatabaseSpecAndSecret(store model.InstallationDatabaseStoreInterface, logger log.FieldLogger) (*mmv1alpha1.Database, *corev1.Secret, error) {
+	installationDatabaseName := MattermostRDSDatabaseName(d.installationID)
+
+	logger = logger.WithField("rds-multitenant-database", installationDatabaseName)
+
+	multitenantDatabases, err := store.GetMultitenantDatabases(&model.MultitenantDatabaseFilter{
```

Comment:
If you create a method `GetMultitenantDatabaseForInstallationID` you can move all the logic to check that there is one, and only one, returned field to that method. 0/5 on needing to do it, but just an idea.